go/worker/keymanager/worker.go

```diff
@@ -408,6 +395,15 @@ func (w *Worker) worker() { // nolint: gocyclo
 					continue
 				}
 
+				// Pre-register to allow replication.
+				w.roleProvider.SetAvailable(func(n *node.Node) error {
```

Comment:
Why did you move this here, shouldn't it be part of `updateStatus`?

Reply:
i suggested that, i think we want to do the "pre-registration" (needed for replication) when the runtime is started and only once - so its a bit more suited here? (while the updateStatus would do that on every status update?)

Reply:
Yeah after thinking about the state transitions, you're right about this only mattering here. But it still would need to check if `w.enclaveStatus` is set. So you can keep it here but add the check to not overwrite existing registrations.

Reply:
Done, but the deadlock persists :thinking: